bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/WorkbookEditorsHandler.java

```diff
@@ -162,10 +164,10 @@ private List<EditorReference> getParts(WorkbenchPage page) {
 	 */
 	private Map<EditorReference, String> generateColumnLabelTexts(List<EditorReference> editorReferences) {
 		Map<EditorReference, String> editorReferenceLabelTexts = new HashMap<>(editorReferences.size());
-		Map<String, List<EditorReference>> collisionsMap = editorReferences.stream()
-				.collect(Collectors.groupingBy(r -> r.getName()));
+		Map<Optional<String>, List<EditorReference>> collisionsMap = editorReferences.stream()
```

Comment:
maybe one should filter out `null` names altogether?

Reply:
Then they would disappear from Ctrl-E?

I guess keeping them is the safest approach for now and just fix the regression introduced by 03277d383e9dcd93a037640c5034a0f9a67fd854 .

Reply:
What do you think about using `Objects.requireNonNullOrElse(r.getName(), "N/A");` if the only problem is that name is null (whatever that the should mean to the user).

Reply:
I think I have a better solution:

The actual intended text to be used in Ctrl-E (if unique), is `ref.getTitle()`, which is already null-safe (returns `""` in that case):
https://github.com/eclipse-platform/eclipse.platform.ui/blob/5b623809b68f069f0df2b6ad27f03af3b0c554f9/bundles/org.eclipse.ui.workbench/Eclipse%20UI/org/eclipse/ui/internal/FilteredTableBaseHandler.java#L821

But in 03277d383e9dcd93a037640c5034a0f9a67fd854 I have inadvertently used `getName()` to populate the `collisionMap`.

It should simply also use `getTitle()`.